misc/inmanta-workon-register.sh

```diff
@@ -204,13 +211,19 @@ function __inmanta_workon_register_deactivate {
         unset -f inmanta >/dev/null 2>&1
         # no need to restore PS1 because virtualenv_deactivate already does that
 
-        if [ -n "${INMANTA_CONFIG_ENVIRONMENT}" ] ; then
+        if [ -n "${_OLD_INMANTA_CONFIG_ENVIRONMENT:-}" ] ; then
+            # Nested environment case: reset INMANTA_CONFIG_ENVIRONMENT to outer env
+            INMANTA_CONFIG_ENVIRONMENT="${_OLD_INMANTA_CONFIG_ENVIRONMENT:-}"
+            export INMANTA_CONFIG_ENVIRONMENT
+            unset _OLD_INMANTA_CONFIG_ENVIRONMENT
+        else
+            # Outermost environment case: simply unset INMANTA_CONFIG_ENVIRONMENT
             unset INMANTA_CONFIG_ENVIRONMENT
         fi
 
         ownership_issues=$(find "$inmanta_env_dir" \! -user "$user" -print -quit)
         if [ -n "$ownership_issues" ]; then
-            echo "WARNING: Some files in the environment are not owned by the $user user. To fix this, run \`chown -R $user '$inmanta_env_dir'\` as root." >&2
+            echo "WARNING: Some files in the environment are not owned by the $user user. To fix this, run \`chown -R $user:$user '$inmanta_env_dir'\` as root." >&2
```

Comment:
From the tests I get the impression that the `inmanta_env_dir` variable doesn't get expanded correctly. I think (not sure) this should do it:
```suggestion
            echo "WARNING: Some files in the environment are not owned by the $user user. To fix this, run \`chown -R $user:$user '\''$inmanta_env_dir'\''\` as root." >&2
```
Iirc the naive approach I suggested first closes the `eval` string we open on line 205, evaluates the variable, then opens a new string and concatenates the three together. Only the variable does not exist in the outer scope. So instead we close the string, concatenate a single quote, and open the string again.